mern-ecommerce/controllers/user.js

```diff
@@ -125,3 +125,19 @@ exports.purchaseHistory = (req, res) => {
       res.json(orders);
     });
 };
+
+exports.deleteUser = (req, res) => {
+  
+  const { userId } = req.params;
+  User.findByIdAndRemove(userId, (err, user) => {
```

Comment:
The `userId` parameter is not validated before being used in the `findByIdAndRemove` method. It should be checked if it exists and is a valid ObjectId before making the database call to prevent potential errors.